src/components/input/input.styles.ts

```diff
@@ -280,4 +280,9 @@ export default css`
   ::-ms-reveal {
     display: none;
   }
+
+  /* Don't show the firefox clear icon in Firefox */
+  .hide-firefox-clear-icon {
+    clip-path: inset(0 2em 0 0);
```

Comment:
Nice. This is the only solution I was able to find in the wild. I agree that the Shoelace clear button should be the one that shows to keep things consistent.